addons/point_of_sale/tests/test_performances.py

```diff
@@ -0,0 +1,30 @@
+from unittest import skipIf
+
+from odoo.addons.point_of_sale.tests.test_frontend import TestPointOfSaleHttpCommon
+
+from odoo.cli.populate import Populate
+from odoo.tests.common import tagged
+from odoo.tools import mute_logger, config
+
+
+@skipIf('pos_performance' not in config['test_tags'], "This test is intended for local performance testing of POS with a large dataset.")
+@tagged('pos_performance', 'post_install', '-at_install')
+class TestPosPerformance(TestPointOfSaleHttpCommon):
+
+    def __populate_model(self, model_name, total_count):
+        before_count = self.env[model_name].search_count([])
+        populate_count = round(total_count / before_count) - 1
+        Populate.populate(self.env, {model_name: populate_count}, 1)
+
+        after_count = self.env[model_name].search_count([])
+        print("====" * 25)
+        print("Before Product Count: %s\nAfter Product Count: %s" % (before_count, after_count))
```

Comment:
```suggestion
        logger.info("Before %s Count: %s\nAfter %s Count: %s" % (model_name, before_count, model_name, after_count))
```

- use logger instead
- make generic for other objects